scipy/stats/morestats.py

```diff
@@ -956,6 +956,8 @@ def boxcox(x, lmbda=None, alpha=None):
 
         If `lmbda` is None, find the lambda that maximizes the log-likelihood
         function and return it as the second output argument.
+    bounds : {None, tuple}, optional
```

Comment:
1. Why `tuple`? It appears that abstractly what  is needed is an object with 2 floats, so that `np.asarray(bounds)` has shape `(2,)`.
I'd suggest describing it as `array_like of length 2` and modifying the implementation to match.

2. How are `bounds` and `brack` related? 
If both are specified are both used?

Reply:
* I defined it as a 2-tuple because `brack` is also defined as a 2-tuple, but I'm happy to change it to array-like and internally wrap it in `np.asarray` 
* `brack` is only used with `optimze.brent` where is serves as a starting interval for a downhill bracket search but the obtained solution is not guaranteed to be in the interval,
* `bounds` is only used with `optimize.fminbound` and guarantees the solution to be in the interval, `fminbound` does not accept `brack` 
* I'm not sure how important `brack` is, we could remove it, alternatively we could raise a warning/error if both are given

Reply:
I wonder if `brack` was made a tuple to avoid a mutable default argument.  The `brent` code only requires something that has a `len` and `__getitem__`.  [The same effect could have been done with `brack=None` in the signature and changing it to `(-2, 2)` or `[-2, 2]` inside if not provided in the function call.]

Thinking (out loud) about the similarity of `brack` and `bounds` arguments, another approach might be to add a `solver={'brent'|'fminbound'}` argument and use `brack` as the interval for both.  I.e. 
  - If `solver='brent'`, the default, use `optimize.brent` as the solver with `brack` the start for the downhill search.    
 - If `solver='fminbound'`, the solver used is `optimize.fminbound`, and `brack` contains the optimization bounds `x1` and `x2` needed for that solver.

That would avoid adding a 2nd similar-but-not-quite-compatible-with-the-first interval to the API.

Any thoughts on the usability and confusability of the two approaches?

Reply:
Adding `solver` as another argument may be confusing, especially because in the end both solvers use the same method, i.e. Brent, just one with bounds and the other one without bounds.

A compromise may be to use the [`minimize_scalar`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize_scalar.html#scipy.optimize.minimize_scalar) interface inside `boxcox_normmax` and expose the relevant input parameters, e.g. 

* `boxcox_normax(x, bracket=None, bounds=None, method='pearsonr', optimizer='brent', tol=None, options=None)`
* `boxcox(x, lmbda=None, bounds=None, alpha=None)` 

This way we have the simplicity of `boxcox` for users who do not want to worry about optimisation specifics and greater flexibility with `boxcox_normmax`

Reply:
For backward compatibility, `bracket` should probably still be named `brack`